app/main.py

```diff
@@ -1 +1,38 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self.__hidden = False
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f'{{Name: {self.name}, '
+                f'Health: {self.health}, '
+                f'Hidden: {self.__hidden}}}')
+
+    def die(self) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+    @property
+    def hidden(self) -> bool:
+        return self.__hidden
+
+    @hidden.setter
+    def hidden(self, value: bool) -> None:
+        self.__hidden = value
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
```

Comment:
The `hide` method toggles the `hidden` property of the `Herbivore` class. While this works, it might be clearer to explicitly set the `hidden` property to `True` when hiding and `False` when revealing. This would make the intention of the method more explicit.